packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/object/services/DefaultContentService.java

```diff
@@ -32,21 +34,28 @@
 @Service
 public class DefaultContentService implements IDefaultContentService {
 
-    private final ComposedAdapterFactory composedAdapterFactory;
+    private final List<Descriptor> composedAdapterFactoryDescriptors;
 
-    public DefaultContentService(ComposedAdapterFactory composedAdapterFactory) {
-        this.composedAdapterFactory = Objects.requireNonNull(composedAdapterFactory);
+    public DefaultContentService(List<Descriptor> composedAdapterFactoryDescriptors) {
+        this.composedAdapterFactoryDescriptors = Objects.requireNonNull(composedAdapterFactoryDescriptors);
     }
+
     @Override
     public List<Object> getContents(Object object) {
         List<Object> contents = new ArrayList<>();
         if (object instanceof EObject eObject) {
-            Adapter adapter = this.composedAdapterFactory.adapt(eObject, IEditingDomainItemProvider.class);
+            var adapterFactories = new ArrayList<>();
+            for (Descriptor composedAdapterFactoryDescriptor : this.composedAdapterFactoryDescriptors) {
+                adapterFactories.add(composedAdapterFactoryDescriptor.createAdapterFactory());
+            }
+            var composedAdapterFactory = new ComposedAdapterFactory(adapterFactories.toArray(AdapterFactory[]::new));
+            Adapter adapter = composedAdapterFactory.adapt(eObject, IEditingDomainItemProvider.class);
```

Comment:
Same remark as above: if possible, use `eObject.eResource().getResourceSet().getAdapterFactories()` instead of creating a new ComposedAdapterFactory on every `getContent()` call.